rclcpp/test/test_node.cpp

```diff
@@ -442,11 +442,13 @@ TEST_F(TestNode, declare_parameters_with_no_initial_values) {
   auto node = std::make_shared<rclcpp::Node>("test_declare_parameters_node"_unq);
   {
     // with namespace, defaults, no custom descriptors, no initial
-    auto values = node->declare_parameters<int>("namespace1", {
+    int64_t bigger_than_int = INT64_MAX - 42;
+    auto values = node->declare_parameters<int64_t>("namespace1", {
       {"parameter_a", 42},
       {"parameter_b", 256},
+      {"parameter_b", bigger_than_int},
     });
-    std::vector<int> expected = {42, 256};
+    std::vector<int64_t> expected = {42, 256, bigger_than_int};
```

Comment:
:thinking: Why was there a warning to begin with?

Reply:
So, parameters are always stored as `int64_t`, and the type in this vector causes the deduced template argument to be the same, and therefore you ended up with some template code that boiled down to `int = int64_t` which triggers the warning.